webui/react/src/hooks/useKeyTracker.ts

```diff
@@ -64,4 +64,39 @@ const useKeyTracker = (): void => {
   }, [handleKeyUp, handleKeyDown]);
 };
 
+export type ShortcutConfig = {
+  altKey: boolean;
+  key: string;
+  metaKey: boolean;
+  shiftKey: boolean;
+  ctrlKey: boolean;
+};
+
+const keyNames = {
+  altKey: 'Alt',
+  ctrlKey: 'Ctrl',
+  metaKey: 'Super',
+  shiftKey: 'Shift',
+};
+
+export const shortcutMatch = (e: KeyboardEvent, shortcut: ShortcutConfig): boolean => {
+  return (
+    e.key.toUpperCase() === shortcut.key.toUpperCase() &&
+    (!shortcut.altKey || e.altKey) &&
+    (!shortcut.metaKey || e.metaKey) &&
+    (!shortcut.shiftKey || e.shiftKey) &&
+    (!shortcut.ctrlKey || e.ctrlKey)
+  );
+};
+
+export const getShortcutString = (shortcut: ShortcutConfig): string => {
+  let shortcutString = '';
+  if (shortcut.altKey) shortcutString += `${keyNames.altKey} +`;
+  if (shortcut.ctrlKey) shortcutString += `${keyNames.ctrlKey} + `;
+  if (shortcut.shiftKey) shortcutString += `${keyNames.metaKey} + `;
+  if (shortcut.metaKey) shortcutString += `${keyNames.shiftKey} + `;
+  if (shortcut.key) shortcutString += shortcut.key.toUpperCase();
+  return shortcutString;
+};
+
```

Comment:
these helpers may be used in further keyboard shortcut-related work

Reply:
Hi John, I was working on pretty much the same utility functions [previously](https://github.com/determined-ai/determined/pull/7147), do you mind rebase upon them. Also feel free to make changes if needed.